src/GitHub.TeamFoundation.14/Services/TeamExplorerServices.cs

```diff
@@ -66,10 +67,15 @@ public void ShowHomePage()
 
         public void ShowPublishSection()
         {
+#if TEAMEXPLORER14 || TEAMEXPLORER15
             var te = serviceProvider.TryGetService<ITeamExplorer>();
             var page = te.NavigateToPage(new Guid(TeamExplorerPageIds.GitCommits), null);
             var publish = page?.GetSection(new Guid(GitHubPublishSection.GitHubPublishSectionId)) as GitHubPublishSection;
             publish?.Connect();
+#else
+            IGitActionsExt2 gitActionsExt = serviceProvider.TryGetService<IGitActionsExt2>();
```

Comment:
Will we need to avoid attempting to load the `IGitActionsExt2` on pre-Git experience versions of VS 2019?

We could do something like this:

```cs
        public void ShowPublishSection()
        {
#if TEAMEXPLORER16
            // Only call InitializeOrPushRepositoryToGitService when IGitActionsExt2 exists
            if (FindIGitActionsExt2() is object)
            {
                InitializeOrPushRepositoryToGitService();
                return;
            }
#endif

            var te = serviceProvider.TryGetService<ITeamExplorer>();
            var page = te.NavigateToPage(new Guid(TeamExplorerPageIds.GitCommits), null);
            var publish = page?.GetSection(new Guid(GitHubPublishSection.GitHubPublishSectionId)) as GitHubPublishSection;
            publish?.Connect();
        }

#if TEAMEXPLORER16
        static Type FindIGitActionsExt2()
        {
            var type = typeof(IGitActionsExt);
            var name = $"{type.FullName}2";
            return type.Assembly.GetType(name, false);
        }

        void InitializeOrPushRepositoryToGitService()
        {
            IGitActionsExt2 gitActionsExt = serviceProvider.TryGetService<IGitActionsExt2>();
            gitActionsExt?.InitializeOrPushRepositoryToGitService();
        }
#endif
```

Reply:
I'm fine with this. I'll make the change.

Reply:
Done.